### PR TITLE
Use a common case sidebar macro and endpoint to view/download all custom case reports

### PR DESCRIPTION
**Pull request by northwestwitch**
The code to view the available case reports and to view them is very repetitive at the moment.

This PR refactors the code so **all case reports belonging to CUSTOM_CASE_REPORTS**:

- Will be **displayed using a common macro** on case sidebar (called in a loop when iterating thru the values of CUSTOM_CASE_REPORTS)
- **Links to the reports will be displayed/downloaded using a common endpoint** based on the cases views (blueprints.cases.views.custom_report)

The advantage of this refactoring is that whenever you add a new element to the CUSTOM_CASE_REPORTS (another type of report), if this report is available for the case, it will be displayed out of the box

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
@fevac this one. Still a draft (haven't tested it on stage yest) and haven't written automatic tests!

**Comment by northwestwitch** [comment]
This is ready for review now!

**Comment by dnil** [comment]
> This is ready for review now!

Super! I’ll have a look in the afternoon!

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:132]: 😸

File: scout/server/blueprints/cases/views.py
Comment (dnil) [inline scout/server/blueprints/cases/views.py:632]: So much better to maintain just one! 🌷

File: tests/server/blueprints/cases/test_cases_views.py
Comment (dnil) [inline tests/server/blueprints/cases/test_cases_views.py:18]: 💯

**Review by dnil — CHANGES_REQUESTED** [review]
Super! Will be so smooth to add new reports! Code looks all ok to me, though admittedly its a little hard to read the git diff. 😊 

Observations so far from testing on local and stage:
- the text captions get a little crowded on the small sidebar. Would e.g. `ms-1` be enough on them? 
Screenshot 2022-11-09 at 14 08 42
- the custom file downloads appear to get their original names, not the new custom `download_name` ones  in Safari? Will test more, but submitting this to speed the check up.. 😁

**Comment by dnil** [comment]
Yeah, seems to be the same in Chrome.. Ah, wait, the tag is `format`not `report_format`, right? 😊

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:145]:
```suggestion
                              case_name=case.display_name, report_type=report_type, report_format='pdf') }}" download><span class="fa fa-file-pdf-o"></span></a>
```

Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:145]: Ah yes, thanks!

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html
Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:132]: Had to do it in this ugly way. Jinja has no case $report_type or something similar..

Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:132]: Huh? I would have naively thought
```suggestion
  {% if case[report_type] %}
```
or better
```suggestion
  {% if case.get(report_type) %}
```
would just work? 😊

Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:132]: We learn each day!

Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:132]: Wait you are right, it is actually working! I don't know what I wrote that didn't work yesterday, but I'll fix this way: it's way easier! 😆

Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html]: Being naive pays off in simplicity! 😄

**Comment by northwestwitch** [comment]
> Observations so far from testing on local and stage:
> 
> * the text captions get a little crowded on the small sidebar. Would e.g. `ms-1` be enough on them?
>   Screenshot 2022-11-09 at 14 08 42

Good suggestion. Fixed and it looks like this now:

image

**Comment by dnil** [comment]
> > Observations so far from testing on local and stage:
> > 
> > * the text captions get a little crowded on the small sidebar. Would e.g. `ms-1` be enough on them?
> >   Screenshot 2022-11-09 at 14 08 42
> 
> Good suggestion. Fixed and it looks like this now:
> 
> image

What do you think about floating the icons right?

**Comment by dnil** [comment]
> > What do you think about floating the icons right?
> 
> I'm trying, but class="float-right" doesn't work, I think because of the style of the parent elements (style of the sidebar)

Ok, no worries. It’s ok as is as well, but could look nice!

**Comment by northwestwitch** [comment]
> Ok, no worries. It’s ok as is as well, but could look nice!

I know. It's not the first time that I try to fix that but there must be some inherited setting from the parent element

**Comment by dnil** [comment]
Hm, ok, the report_format is now working ok, but sorry, I'm getting a couple of more errors/weridness now. Both are with reports in formats we haven't really had before.

MutiQC convereted to pdf is not very pretty. 20 pages of "loading..". 😊 
Screenshot 2022-11-09 at 16 32 58

The other is with trying to dl an arriba gene fusion report as pdf, which it is already, and hence:
Screenshot 2022-11-09 at 16 30 52

Perhaps we need another key or two for the CUSTOM_REPORT, giving the native format and permissive conversions?

**Comment by dnil** [comment]
> > Ok, no worries. It’s ok as is as well, but could look nice!
> 
> I know. It's not the first time that I try to fix that but there must be some inherited setting from the parent element

Those can be a pain.. Not worth spending much time on - perhaps it should be an `data-bs-toggle="offcanvas"` anyway..

**Comment by northwestwitch** [comment]
> Hm, ok, the report_format is now working ok, but sorry, I'm getting a couple of more errors/weridness now. Both are with reports in formats we haven't really had before.

No worries, it's actually very good that you checked, because I didn't! :) I'll fix it tomorrow!

**Comment by northwestwitch** [comment]
> Perhaps we need another key or two for the CUSTOM_REPORT, giving the native format and permissive conversions?

Done now. Take a look when you have time!

**Review by dnil — APPROVED** [review]
Looks great and tests ok on my local. Can't test well on stage as it is with report storage atm. I'll approve: up to you if you want to merge and test more before release, or wait a bit with merging until stage is healthy again!

**Comment by northwestwitch** [comment]
> Looks great and tests ok on my local. Can't test well on stage as it is with report storage atm. I'll approve: up to you if you want to merge and test more before release, or wait a bit with merging until stage is healthy again!

I think I'll just merge. We have to re-test it before the new release anyway! Thanks for the review!

**Pull request merged**
